Marshal/Marshal.swift

```diff
@@ -0,0 +1,187 @@
+//
+//  M A R S H A L
+//
+//       ()
+//       /\
+//  ()--'  '--()
+//    `.    .'
+//     / .. \
+//    ()'  '()
+//
+//
+
+
+import Foundation
+
+
+public protocol Marshal {
+    subscript(key: KeyType) -> Any? { get }
+    func anyForKey(key: KeyType) throws -> Any
+    func valueForKey<A: ValueType>(key: KeyType) throws -> A
+    func valueForKey<A: ValueType>(key: KeyType) throws -> A?
```

Comment:
Why are `anyForKey` and `valueForKey` (in all of its forms) part of the protocol declaration? Shouldn't the protocol be only `subscript`, and all of the rest be in the extension only?